drivers/pcie/host/shell.c

```diff
@@ -29,7 +29,34 @@ static void show_msi(const struct shell *shell, pcie_bdf_t bdf)
 	msi = pcie_get_cap(bdf, PCIE_MSIX_CAP_ID);
 
 	if (msi) {
-		shell_fprintf(shell, SHELL_NORMAL, "    MSI-X support\n");
+		uint32_t offset, table_size;
+		uint8_t bir;
+
+		data = pcie_conf_read(bdf, msi + PCIE_MSIX_MCR);
+
+		table_size = ((data & PCIE_MSIX_MCR_TSIZE) >>
+			      PCIE_MSIX_MCR_TSIZE_SHIFT) + 1;
+
+		shell_fprintf(shell, SHELL_NORMAL,
+			      "    MSI-X support%s table size %d\n",
```

Comment:
I think this should be `MSI-X support %s, table size %d\n` instead.

Also, should it be showing table size if MSI-X is disabled?

Reply:
Why not? it is the same in lspci

Reply:
Because it looks less strange/more readable. I.e. please fix it in lspci as well (separate patch or PR though)

Reply:
I mean lspci in Linux. I think it does make sense to print this information even if it is disabled.

Reply:
> I mean lspci in Linux. I think it does make sense to print this information even if it is disabled.

I dont think anyone requested not to print something. My comment was just about how to split up the format string.